### ISSUE TITLE
Markdown layer clean-org-tables will unexpectedly change ditta code

### ISSUE THREAD

**Issue by qiuchengxuan** [syl20bnr/spacemacs#14774]
#### Description :octocat:
Embedded ditta code block unexpectedly changed

`cleanup-org-tables-on-save` should not be applied to code blocks, or should able to be disabled

#### Reproduction guide :beetle:
- Start Emacs
- Write a markdown file with [ditta][ditta] code block
- Save file

[ditta]: https://plantuml.com/en/ditaa

*Observed behaviour:* :eyes: :broken_heart:
ditta unexpectedly changed

*Expected behaviour:* :heart: :smile:
ditta should not be changed

#### System Info :computer:

N/A

#### Backtrace :paw_prints:

N/A

**Comment by lebensterben** [comment]
Confirmed.

**Comment by smile13241324** [comment]
I will have a look at this in the next few day.

**Comment by smile13241324** [comment]
I have removed `cleanup-org-tables-on-save` completely for its implementation was replacing `-+-` in the entire buffer which was causing a lot of side effects for almost all markdown use-cases.

The reason this function has been added in the first place was that there was no independent table implementation for markdown yet. As a workaround Spacemacs tried to use `orgtbl-mode` tables instead which need to be changed to pass as markdown tables. Today markdown has its own table editor function which is now bound to `SPC m i T` making the entire org table workaround obsolete.

I have kept `orgtbl-mode` in the layer for now to avoid breaking old configs which may have org table insert be bound to a local binding, still markdown-layer itself does not longer bind any table related commands from `orgtbl-mode`.
